### DISCUSSION TITLE
How to get a path as dynamic parameter in route?

### DISCUSSION THREAD

**Discussion by subhamcodes** [adonisjs/core#2289] [Help]
I am trying to build something where I need to pass the path to a file in route parameters. So far I can use * in a route like 
`
Route.get('/files/*', async({ params }) => { return params })
`
Then in my edge file I am doing `{{ route('name') }}/abcd/123/a.jpg`
It's working so far and I can get the path however it's also adding * in the rendered source code of the view. How to get rid of the * from the source code and is there any way to do it like Express -` '/files/:path*' `and then I can use the path as a dynamic parameter.

**Comment by McSneaky** [comment]
You can use `Route.get('/files/:name', async ({params}) => params)

Then `:name` will be the dynamic part
https://preview.adonisjs.com/guides/http/routing#dynamic-urls
